### PR TITLE
Removing the `wire_map` argument from `qml.is_commuting`

### PR DESCRIPTION
**Pull request by PietropaoloFrisoni**
**Context:** ***Part of  deprecations and removals for pennylane-0.37***

**Description of the Change:** Removed the `wire_map` argument from `qml.is_commuting`.

[sc-58988]

### CODE REVIEW

File: doc/releases/changelog-dev.md
Comment (PietropaoloFrisoni) [inline doc/releases/changelog-dev.md:1]: As mentioned on Shortcut, it has been decided not to go through a deprecation cycle because the presence of `wire_map` is not functional and can be viewed more as a bug than a feature. 

Therefore, the `doc/development/deprecations.rst` file has not been modified in this PR.

**Comment by trbromley** [comment]
@glassnotes @josh146, looking at the commit history you were both involved in adding `qml.is_commuting`. Do you know why we added a `wire_map`? Our thinking has been that a mapping of wires does not alter whether two operations commute, but I want to check if we're missing something.

**Comment by josh146** [comment]
@trbromley I'm sorry, I don't recall the reasoning here :( Do you have a link to the original PR?

File: pennylane/ops/functions/is_commuting.py
Comment (albi3ro) [inline pennylane/ops/functions/is_commuting.py:29]:
```suggestion
        pauli_word_1 (Operator): first Pauli word in commutator.
        pauli_word_2 (Operator): second Pauli word in commutator.
```

Might as well update this now.

**Comment by glassnotes** [comment]
> @glassnotes @josh146, looking at the commit history you were both involved in adding `qml.is_commuting`. Do you know why we added a `wire_map`? Our thinking has been that a mapping of wires does not alter whether two operations commute, but I want to check if we're missing something.

Yeah I think it was added to support taking the symplectic inner product to evaluate commutativity, and the functions that convert to binary symplectic rep required a wire order.

**Comment by PietropaoloFrisoni** [comment]
Thanks everyone. I'll merge this PR as soon as I have a second approval!

File: doc/releases/changelog-dev.md
Comment (astralcai) [inline doc/releases/changelog-dev.md:68]: I don't think we need to justify the change in the changelog @trbromley ?

**Review by astralcai — COMMENTED** [review]
Don't forget to update the `deprecations` page as well

**Comment by PietropaoloFrisoni** [comment]
@astralcai I left a comment about that above. I'll repost it here;

As mentioned on Shortcut, it has been decided not to go through a deprecation cycle because the presence of `wire_map` is not functional and can be viewed more as a bug than a feature.

Therefore, the `doc/development/deprecations.rst` file has not been modified in this PR.

File: doc/releases/changelog-dev.md
Comment (albi3ro) [reply doc/releases/changelog-dev.md:68]: Given these types of things could irritate or confuse someone, it's good to justify the change being worth it.

**Review by astralcai — APPROVED** [review]

**Pull request merged**
